llvm/lib/Transforms/Utils/LowerMemIntrinsics.cpp

```diff
@@ -456,6 +456,56 @@ static void createMemMoveLoop(Instruction *InsertBefore, Value *SrcAddr,
   ElseTerm->eraseFromParent();
 }
 
+static void createMemSetPatternLoop(Instruction *InsertBefore, Value *DstAddr,
+                                    Value *Count, Value *SetValue,
+                                    Align DstAlign, bool IsVolatile) {
+  BasicBlock *OrigBB = InsertBefore->getParent();
+  Function *F = OrigBB->getParent();
+  const DataLayout &DL = F->getDataLayout();
+
+  if (DL.isBigEndian())
+    report_fatal_error("memset.pattern expansion not currently "
+                       "implemented for big-endian targets",
+                       false);
```

Comment:
Why this check? Your implementation is endian-ness independent. The endian-specific handling is inside the store.

Reply:
Now removed - this was a holdover from an earlier implementation.